src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -479,6 +484,26 @@ private function addFlashMessageProductV2IsDisabled(): void
         );
     }
 
+    /**
+     * @param int $productId
+     *
+     * @return Response
+     */
+    private function renderMissingAssociation(int $productId): Response
+    {
+        //@todo this error message should be improved to indicate which shop can/should be used for this product and/or how to associate it to the current context
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Product/disabled.html.twig', [
+            'errorMessage' => $this->trans(
+                'This product is not associated to the store selected in the current context, please select another one.',
```

Comment:
```suggestion
                'This product is not associated with the store selected in the multistore header, please select another one.',
```
I just have a doubt about "multistore header" as it's not available on all pages. In any case, we should replace "associated to" with "associated with". :-)

Reply:
It is available on this page when multishop is active so it seems all good